tests/framework/components/element/test_textelement.html

```diff
@@ -0,0 +1,221 @@
+        <script type="text/javascript" src="../../../../src/framework/components/zone/data.js"></script>
+        <script type="text/javascript" src="../../../../src/framework/entity.js"></script>
+        <script type="text/javascript" src="../../../../src/resources/loader.js"></script>
+        <script type="text/javascript" src="../../../../src/resources/animation.js"></script>
+        <script type="text/javascript" src="../../../../src/resources/audio.js"></script>
+        <script type="text/javascript" src="../../../../src/resources/cubemap.js"></script>
+        <script type="text/javascript" src="../../../../src/resources/json.js"></script>
+        <script type="text/javascript" src="../../../../src/resources/material.js"></script>
+        <script type="text/javascript" src="../../../../src/resources/model.js"></script>
+        <script type="text/javascript" src="../../../../src/resources/script.js"></script>
+        <script type="text/javascript" src="../../../../src/resources/text.js"></script>
+        <script type="text/javascript" src="../../../../src/resources/binary.js"></script>
+        <script type="text/javascript" src="../../../../src/resources/texture.js"></script>
+        <script type="text/javascript" src="../../../../src/resources/html.js"></script>
+        <script type="text/javascript" src="../../../../src/resources/css.js"></script>
+        <script type="text/javascript" src="../../../../src/resources/shader.js"></script>
+        <script type="text/javascript" src="../../../../src/resources/scene.js"></script>
+        <script type="text/javascript" src="../../../../src/resources/hierarchy.js"></script>
+        <script type="text/javascript" src="../../../../src/resources/scene-settings.js"></script>
+        <script type="text/javascript" src="../../../../src/resources/folder.js"></script>
+        <script type="text/javascript" src="../../../../src/resources/font.js"></script>
+        <script type="text/javascript" src="../../../../src/resources/texture-atlas.js"></script>
+        <script type="text/javascript" src="../../../../src/resources/sprite.js"></script>
+        <script type="text/javascript" src="../../../../src/resources/parser/json-model.js"></script>
+        <script type="text/javascript" src="../../../../src/resources/parser/scene.js"></script>
+        <script type="text/javascript" src="../../../../src/asset/asset.js"></script>
+        <script type="text/javascript" src="../../../../src/asset/asset-variants.js"></script>
+        <script type="text/javascript" src="../../../../src/asset/asset-registry.js"></script>
+        <script type="text/javascript" src="../../../../src/backwards-compatibility.js"></script>
+        <script type="text/javascript" src="../../../../src/scene/immediate.js"></script>
+        <script type="text/javascript" src="../../../../src/scene/lightmapper.js"></script>
+        <script type="text/javascript" src="../../../../src/scene/batching.js"></script>

```

Comment:
@daredevildave obviously this isn't good :) I'm basically duplicating `dependencies.txt`. However, I couldn't see a way to test `Elements` or `Entities` without including most of these dependencies. Perhaps we could have a way to transform `dependencies.txt` to a list of `<script>` includes and inject it when loading the tests in a browser?